### PR TITLE
Add improvement of allowing specific sinks and sources to be connected

### PR DESCRIPTION
**Pull request by dnwick**
## Purpose
Allow http request , http response sinks to connect to http-response and http-request sources

## Goals
Should be able to connect particular sinks to sources

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/EdgesGenerator.java [hunk 1]
```diff
@@ -533,4 +539,15 @@ private NodeType getSiddhiElementType(SiddhiElementConfig siddhiElementConfig) t
     private static String generateEdgeId(String parentID, String childID) {
         return String.format("%s_%s", parentID, childID);
     }
+
+    private SiddhiElementConfig getElementWithSinkCorrelateId(String correlateId) {
+        for (SourceSinkConfig sourceConfig : siddhiAppConfig.getSourceList()) {
+            if (sourceConfig.getCorrelateId() != null) {
+                if (sourceConfig.getCorrelateId().equals(correlateId)) {
```
Comment (mohanvive) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/EdgesGenerator.java]: Do you need two levels of if condition for this? We can merge this together

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/SourceSinkConfigsGenerator.java [hunk 2]
```diff
@@ -118,8 +124,13 @@ private SourceSinkConfig generateSourceOrSinkConfig(SourceOrSinkAnnotation annot
                 options.add(annotationConfigGenerator.generateAnnotationConfig(sourceOrSinkAnnotation));
             }
         }
+        boolean correlateIdExist = false;
+        if (correlateId != null) {
```
Comment (mohanvive) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/SourceSinkConfigsGenerator.java]: Use a ternary operator for ths.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js [hunk 3]
```diff
@@ -450,6 +443,19 @@ define(['require', 'log', 'jquery', 'backbone', 'lodash', 'designViewUtils', 'dr
                         } else {
                             return connectionValidity = true;
                         }
+                    } //we allowing all sinks to be connected to source here if connections points are available to connect.
+                      //This need to be changed if there are use cases to allow specifics to connect
+                    else if (targetElement.hasClass(constants.SOURCE)) {
+                        if (!sourceElement.hasClass(constants.SINK)) {
```
Comment (mohanvive) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js]: Invert the if condition and reduce the nested if clauses.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java [hunk 4]
```diff
@@ -64,4 +65,14 @@ public String getType() {
     public MapperConfig getMap() {
         return map;
     }
+
+    public boolean isCorrelateIdExist() {
+
```
Comment (mohanvive) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java]: Follow the standard across the code. Either 1st line of the method should be a line space but it seems there are some methods without any line space. As a best practice avoid have a line space as the 1st line in a method.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java [hunk 5]
```diff
@@ -32,17 +32,18 @@
     private String type;
     private List<String> options;
     private MapperConfig map;
+    private boolean correlateIdExist;
```
Comment (mohanvive) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java]: A better name would be isCorrelationIdExists.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java [hunk 6]
```diff
@@ -32,17 +32,18 @@
...
+    private String correlateId;
```
Comment (mohanvive) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java]: Can we rename this as correlationId.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/EdgesGenerator.java (hunk 1 above)
Comment (dnwick) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/EdgesGenerator.java]: Fixed

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/SourceSinkConfigsGenerator.java (hunk 2 above)
Comment (dnwick) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/SourceSinkConfigsGenerator.java]: Fixed

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java (hunk 4 above)
Comment (dnwick) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java]: fixed

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java (hunk 5 above)
Comment (dnwick) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java]: fixed

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java (hunk 6 above)
Comment (dnwick) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/sourcesink/SourceSinkConfig.java]: fixed

**Pull request merged**